lazy_theta_star_b/src/lazy_theta_star_b.cpp

```diff
@@ -0,0 +1,529 @@
+                                   std::shared_ptr<nav2_costmap_2d::Costmap2DROS> costmap_ros) {
+        node_ = parent;
+        name_ = name;
+        tf_ = tf;
+        costmap_ = costmap_ros->getCostmap();
+        global_frame_ = costmap_ros->getGlobalFrameID();
+
+        nav2_util::declare_parameter_if_not_declared(
+                node_, name_ + ".how_many_corners", rclcpp::ParameterValue(
+                        8));
+
+        node_->get_parameter(name_ + ".how_many_corners", how_many_corners_);
+
+        if (how_many_corners_ != 8)
+            how_many_corners_ = 4;
+
+        nav2_util::declare_parameter_if_not_declared(
+                node_, name_ + ".interpolation_dist", rclcpp::ParameterValue(
+                        0.1));
+
+        node_->get_parameter(name_ + ".interpolation_dist", interpolation_dist);
+
+    }
+
+    void LazyThetaStarB::cleanup() {
+        RCLCPP_INFO(
+                node_->get_logger(), "CleaningUp plugin %s of type LazyThetaStarB",
+                name_.c_str());
+    }
+
+    void LazyThetaStarB::activate() {
+        RCLCPP_INFO(
+                node_->get_logger(), "Activating plugin %s of type LazyThetaStarB",
+                name_.c_str());
+    }
+
+    void LazyThetaStarB::deactivate() {
+        RCLCPP_INFO(
+                node_->get_logger(), "Deactivating plugin %s of type LazyThetaStarB",
+                name_.c_str());
+    }
+
+    nav_msgs::msg::Path LazyThetaStarB::createPlan(
+            const geometry_msgs::msg::PoseStamped &start,
+            const geometry_msgs::msg::PoseStamped &goal) {
+
+
+           RCLCPP_INFO(
+                node_->get_logger(), "GOT THE SOURCE AND DESTINATION ------ (%f, %f) && (%f, %f)",
+               start.pose.position.x, start.pose.position.y, goal.pose.position.x, goal.pose.position.y);
+
+        costmap_->worldToMap(start.pose.position.x, start.pose.position.y, reinterpret_cast<unsigned int &>(src.x),
+                             reinterpret_cast<unsigned int &>(src.y));
+
+
+        costmap_->worldToMap(goal.pose.position.x, goal.pose.position.y, reinterpret_cast<unsigned int &>(dst.x),
+                             reinterpret_cast<unsigned int &>(dst.y));
+        src.x = int(src.x);
+        src.y = int(src.y);
+
+        dst.x = int(dst.x);
+        dst.y = int(dst.y);
+
+        data.clear();
+        pq.clear();
+        path.clear();
+        posn.clear();
+        lethal_cost = LETHAL_COST;
+
+        sizeX = costmap_->getSizeInCellsX();
+        sizeY = costmap_->getSizeInCellsY();
+
+        //has been added so that if the robot is stuck at a place with costmap cost higher than 127, then only for that run, the planner be able to give a path
+        if (costmap_->getCost(src.x, src.y) >= lethal_cost && costmap_->getCost(src.x, src.y) <= 253) {
+            lethal_cost = costmap_->getCost(src.x, src.y);
+        }
+        
+        initializePosn();
+        data.reserve(int((sizeX) * (sizeY) * 0.01));
```

Comment:
Why the 0.01?

Reply:
Since the number number of cells, would be much lower than the required amount for almost all cases.

Reply:
I suppose I'd argue not to set it at all, but I guess this is fine. 

Something I do see though is that you do alot of `data[curr_id]` throughout the algorithm. You should at the start of each iteration do a `DataT & curr_data = data[curr_id]` so that you only do the look up once, right now you're doing dozens of look ups every iteration in the vector.